man/inforevealer.1
.\" Process this file with
.\" groff -man -Tascii foo.1
.\"
.TH INFOREVEALER 1 "JULY 2010" INFOREVEALER "User Manual"
.SH NAME
inforevealer \- generates a file containing useful information about the system.
.SH SYNOPSIS
.B inforevealer [-p] [-c
.I category
.B ]
.\".I file
.\".B ...
.SH DESCRIPTION
.B inforevealer
provide easily a check-up of useful commands, logfiles or configuration files through categories. This check-up might be used by people who experiences some troubles with their distributions on forums, IRC, jabber... Thus, contributors could identify quickly problems thanks to relevant information. 
.SH OPTIONS
.IP \-c,\ \-\-category=CATEGORY
Use the category CATEGORY.
.IP \-f,\ \-\-file=FILE
Write the report in FILE.
.IP \-l,\ \-\-list
Print the list of categories.
.IP \-p,\ \-\-pastebin
Send the report on a pastebin website.
.IP \-w,\ \-\-website=WEBSITE
Use the pastebin WEBSITE.
.IP \-\-verbose
Use the verbose mode.
.IP \-h,\ \-\-help
Print a reminder of options.
.IP \-\-gui
Start the graphic interface. Other options are ignored.
.SH FILES
.I /etc/inforevealer.d/categories.conf
.RS
The configuration file which contains categories and their contents. This file should respect
.BR /etc/inforevealer.d/validator.conf
specifications.
.RE
.I /etc/inforevealer.d/pastebin/
.RS
This directory owns configuration files for pastebin websites.
.SH COPYRIGHT
(C) Francois Boulogne <fboulogne at april dot org>
.PP This software is distributed under the GNU GPLv2 license.
This is free software: you are free to change and redistribute it.  There is NO WARRANTY, to the extent permitted by law.
.SH AUTHOR
Francois Boulogne <fboulogne at april dot org>


CHANGELOG
VERSION 0. 
RELEASE 2010--

New features:
	* Add a pipe option (only) for commands.

Improvements:
	* Better title style
	* Description of content in a subtitle

################
VERSION 0.5.1 
RELEASE 2010-08-24

improvements:
	* Remove yes/no dialog for password (seems to be confusing)
	* grab enter key for password dialog window
	* better look&feel for the first window

################
VERSION 0.5.0
RELEASE 2010-08-23

new features:
	* inspect directories
	* i18n on both source and config files
	* inforevealer.desktop added
	* man page
	* Makefile/script to (un)install 
bugfix:
	* Keep the main window open
	* center windows
	* Opensuse installs sudo without configuration: try su before sudo
	* correction open-suse release file

################
VERSION 0.4.1
RELEASE 2010-08-18


Bugfix:
    * bugfix: sudo

################
VERSION 0.4 
RELEASE 2010-08-17

new features:
    * GUI
    * Allow file=~/foobar in configfile. (~==$HOME)

Bugfix:
    * configfile validator

################
VERSION 0.3 
RELEASE 2010-07-31

Bugfixes:
	* error in sudo command
	* path troubles for some distributions when runned as root
	* Be more specific for the configobj dependance
	* Corrections for debian(-like) package config. files

################
VERSION 0.2
RELEASE 2010-07-30

new features:
	* add configuration files
	* switch to root if needed and desired

################
VERSION 0.1 
RELEASE 2010-07-25

First version
features:
	* generate log on harddrive
	* send this log on pastebin


src/inforevealer.py
#!/usr/bin/python
# -*- coding: utf-8 -*-


# Inforevealer
# Copyright (C) 2010  Francois Boulogne <fboulogne at april dot org>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

import io #outputs...
import readconf #read categories
import action # main part...

import sys, time, gettext, string

from pastebin import sendFileContent


gettext.textdomain('inforevealer')
_ = gettext.gettext

__version__="devel"

#####################
#Main part
#####################
def main(argv):
	try:
		import getopt
		import sys

		

		#set default
		dumpfile='/tmp/inforevealer'
		tmp_configfile="/tmp/inforevealer_tmp.conf" #tmp configuration file (substitute)
		verbosity=False
		category=""
		runfile=None #option only for internal use, see above
		gui=False #run the GUI

		defaultPB = "http://pastebin.com" #Default pastebin
		website = defaultPB
		pastebin_choice=False


		#####################
		# GETOPT
		#####################
		try:	
			options, remainder = getopt.gnu_getopt(sys.argv[1:], 'hlc:vf:pw:', ['help',
									   'list',
									   'category=',
									   'verbose',
									   'file=',
									   'pastebin',
									   'website',
									   'runfile=',
									   'gui'
									 ])
									 
		except getopt.GetoptError:
			sys.stderr.write(_("Invalid arguments."))
			io.usage()
			sys.exit(1)

		for opt, arg in options:
			if opt in ('-h', '--help'):
				io.usage()
				sys.exit()
			elif opt in ('-l', '--list'):
				#find categories.conf
				filename=readconf.find_categories_conf()
				#find validator.conf
				spec_filename=readconf.find_validator_conf()
				#open categories.conf with validator.conf
				configfile=readconf.open_config_file(filename,spec_filename)
				# load the list of categories
				list_category=readconf.LoadCategoryList(configfile)
				io.list(list_category)
				sys.exit()
			elif opt in ('-c', '--category'):	
				category=arg
			elif opt in ('-v','--verbose'):
				verbosity=True
			elif opt in ('-f','--file'):
				dumpfile=arg
			elif opt in ('-p','--pastebin'):
				pastebin_choice=True
			elif opt in ('-w','--website'):
				website=arg
				if not website.endswith("/"):
					website += "/"
			elif opt in ('--runfile'):
				runfile=arg
			elif opt in ('--gui'):
				gui=True

		#First to do: runfile (internal use)
		if runfile != None:
			readconf.ReadAndMakeInternalDesire(tmp_configfile)
			sys.exit()
		else:
			#find categories.conf
			filename=readconf.find_categories_conf()
			#find validator.conf
			spec_filename=readconf.find_validator_conf()
			#open categories.conf with validator.conf
			configfile=readconf.open_config_file(filename,spec_filename)
			# load the list of categories
			list_category=readconf.LoadCategoryList(configfile)
			
			if gui==True:
				import gui
				gui.main(configfile,list_category)
			#check if category is ok
			elif category in list_category:
				action.action(category,dumpfile,configfile,tmp_configfile,verbosity)
				sendFileContent(dumpfile,title=category,website=website,version=None)
			else:
				sys.stderr.write(_('Error: Wrong category'))
				io.usage()
				sys.exit(1)
	
	except KeyboardInterrupt:
		sys.exit("KeyboardInterrupt caught.")



#####################
# run main
#####################
if __name__ == "__main__":
	    main(sys.argv[1:])
        

